libraries/AP_IOMCU/AP_IOMCU.cpp

```diff
@@ -789,6 +830,70 @@ void AP_IOMCU::set_brushed_mode(void)
     rate.brushed_enabled = true;
 }
 
+#if HAL_DSHOT_ENABLED
+// set output mode
+void AP_IOMCU::set_dshot_period(uint16_t period_us, uint8_t drate)
+{
+    dshot_rate.period_us = period_us;
+    dshot_rate.rate = drate;
+    trigger_event(IOEVENT_SET_DSHOT_PERIOD);
+}
+
+// set output mode
+void AP_IOMCU::set_telem_request_mask(uint32_t mask)
+{
+    page_dshot dshot {
```

Comment:
I don't consider this a blocker for merging btw, just seems extravagant in bandwidth

Reply:
page_dshot is only 10 bytes and only contains the elements that need to be managed by a command queue to prevent data loss - so does not seem too bad. Could be split out but then need two command queues - more flash etc